drivers/storage/portworx/deployment.go

```diff
@@ -0,0 +1,670 @@
+package portworx
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/google/shlex"
+	corev1alpha1 "github.com/libopenstorage/operator/pkg/apis/core/v1alpha1"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	pxContainerName          = "portworx"
+	pxServiceAccount         = "px-account"
+	annotationIsDocker       = "portworx/is-docker"
+	annotationIsPxDev        = "portworx/is-px-dev"
+	annotationRunOnMaster    = "portworx/run-on-master"
+	annotationIsPKS          = "portworx/is-pks"
+	annotationIsCoreOS       = "portworx/is-coreos"
+	annotationIsOpenshift    = "portworx/is-openshift"
+	annotationLogFile        = "portworx/log-file"
+	annotationCustomRegistry = "portworx/custom-registry"
+	annotationCSIVersion     = "portworx/csi-version"
+	annotationMiscArgs       = "portworx/misc-args"
+	templateVersion          = "v4"
+	defaultStartPort         = 9001
+	csiBasePath              = "/var/lib/kubelet/plugins/com.openstorage.pxd"
+	masterTaint              = "node-role.kubernetes.io/master"
+)
+
+type volumeInfo struct {
+	name             string
+	hostPath         string
+	mountPath        string
+	readOnly         bool
+	mountPropagation *v1.MountPropagationMode
+	hostPathType     *v1.HostPathType
+	pks              *pksVolumeInfo
+	openshift        *openshiftVolumeInfo
+	coreos           *coreosVolumeInfo
+}
+
+type pksVolumeInfo struct {
```

Comment:
Had made 3 separate types because there could be overlaps in future. Anyway removed coreos and openshift types because they were needed only when running as docker. For runc it is the same mounts with or without the options enabled.